src/foam/dao/EasyDAO.js

```diff
@@ -102,20 +107,41 @@ foam.CLASS({
         @private */
       name: 'delegate',
       javaFactory: `
-//foam.nanos.logger.Logger logger = (foam.nanos.logger.Logger) getX().get("logger");
+//Logger logger = (foam.nanos.logger.Logger) getX().get("logger");
 //logger.info(this.getClass().getSimpleName(), "delegate", "NSpec.name", getNSpec().getName(), "of_", of_); //Thread.currentThread().getName());
 foam.dao.DAO delegate = getInnerDAO() == null ?
   new foam.dao.MDAO(getOf()) :
   getInnerDAO();
 
 if ( delegate instanceof foam.dao.MDAO ) {
   setMdao((foam.dao.MDAO)delegate);
+  if ( getIndex() != null &&
+       getIndex().length > 0 ) {
+    // IMPORTANT: Indexes must be added before JDAO replay occurs.
```

Comment:
I don't think this is true, but it doesn't hurt.

Reply:
At the moment this is the case, DAOs are simply broken otherwise.